src/fastly.js

```diff
@@ -5,80 +5,93 @@ const program = require(`commander`);
+
+const fastlyServiceIdDescription = `Fastly service ID`;
+const fastlyServiceIdOption = `-s, --serviceid <required>`;
 const fastlyServiceIdErrorMessage = `[error] ${fastlyServiceIdDescription} is required.`;
 
+const fastlyHardPurgeOption = `-h, --hardpurge`;
+const fastlyHardPurgeDescription = `Hard purge immediately; do not use soft purge option.`;
+
 program
   .version(require(`../package.json`).version)
-  .option(`-k, --apikey <apikey>`, `${fastlyApiKeyDescription}.`)
-  .option(`-s, --serviceid <serviceid>`, `${fastlyServiceIdDescription}.`)
-  .option(`-h, --hardpurge`, `Hard purge immediately; do not use soft purge option.`);
 
 program
   .command(`purge-all`)
   .description(`Purge all Fastly content.`)
   .alias(`pa`)
-  .action(() => {
-    if (!fastlyApiKeyPresent(program) || !fastlyServiceIdPresent(program)) {
+  .option(fastlyApiKeyOption, `${fastlyApiKeyDescription}.`)
+  .option(fastlyServiceIdOption, `${fastlyServiceIdDescription}.`)
+  .option(fastlyHardPurgeOption, fastlyHardPurgeDescription)
+  .action((options) => {
+    if (!fastlyApiKeyPresent(options) || !fastlyServiceIdPresent(options)) {
       return;
     }
-    const fastly = require(`fastly`)(program.apikey);
+    const fastly = require(`fastly`)(options.apikey);
 
-    if (program.hardpurge) {
-      fastly.purgeAll(program.serviceid, handleFastlyResponse(`All content purged.`));
+    if (options.hardpurge) {
+      fastly.purgeAll(options.serviceid, handleFastlyResponse(`All content purged.`));
     }
     else {
-      fastly.softPurgeAll(program.serviceid, handleFastlyResponse(`All content purged.`));
+      fastly.softPurgeAll(options.serviceid, handleFastlyResponse(`All content purged.`));
     }
   });
 
 program
   .command(`purge`)
   .description(`Purge content at specified specified URL.`)
+  .option(fastlyApiKeyOption, `${fastlyApiKeyDescription}.`)
+  .option(fastlyServiceIdOption, `${fastlyServiceIdDescription}.`)
+  .option(fastlyHardPurgeOption, fastlyHardPurgeDescription)
   .arguments(`<url>`)
-  .action((url) => {
-    if (!fastlyApiKeyPresent(program) || !fastlyServiceIdPresent(program)) {
+  .action((url, options) => {
+    if (!fastlyApiKeyPresent(options) || !fastlyServiceIdPresent(options)) {
       return;
     }
-    const fastly = require(`fastly`)(program.apikey);
+    const fastly = require(`fastly`)(options.apikey);
 
-    if (program.hardpurge) {
-      fastly.purge(program.serviceid, url, handleFastlyResponse(`Purged URL: ${url}`));
+    if (options.hardpurge) {
+      fastly.purge(options.serviceid, url, handleFastlyResponse(`Purged URL: ${url}`));
     }
     else {
-      fastly.softPurge(program.serviceid, url, handleFastlyResponse(`Purged URL: ${url}`));
+      fastly.softPurge(options.serviceid, url, handleFastlyResponse(`Purged URL: ${url}`));
     }
   });
 
 program
   .command(`purge-key`)
   .description(`Purge content with specified key.`)
```

Comment:
Could we clarify what type of “key” this is? Since we use “API keys” in this cli, I think it would be best to be super explicit about the fact that this is referring to something else (if that’s the case).

On the other hand, if this _is_ the same key, why do we require it as an argument if it’s already required with the `-k` flag?